src/angular-google-analytics.js

```diff
@@ -596,12 +673,23 @@
           });
           _analyticsJs(function () {
             if (that._enhancedEcommerceEnabled()) {
-              $window.ga('ec:addPromo', {
-                id: productId,
-                name: name,
-                creative: creative,
-                position: position
-              });
+              if (angular.isArray(accountId)) {
```

Comment:
This code is copy-pasted a lot. Please find a better way to encapsulate this. With the addition of the _ga function it probably makes sense to perform this conditional logic in that location, which will clean-up all of these other spots.